src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -227,6 +223,14 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
     const navigationFocusControl = useFocusControl(navigationOpen);
     const splitPanelFocusControl = useSplitPanelFocusControl([splitPanelPreferences, splitPanelOpen]);
 
+    const onNavigationToggle = useCallback(
```

Comment:
Moved this handler below because it relies on navigationFocusControl, which is declared earlier

Reply:
We should use our `useStableCallback` util.

There is no promise that `onNavigationChange` is a stable callback, so your code does not really do what you expect.

To an extent, the same applies to other `useCallback` additions here. Make sure their dependencies are really stable (`drawers` array, for example is not)